popde/density_estimate.py

```diff
@@ -120,6 +120,70 @@ def prepare_data(self):
         else:
             self.kde_data = self.std_data
 
+
+    def evaluate_with_transf(self, points):
+        """
+        Transforms the evaluate points based on specified transformations
+        , evaluates the kernel density estimate (KDE) on the transformed points
+         and adjusts the KDE values using the Jacobian of the transformations.
+
+        Parameters:
+        -----------
+        points : array-like, shape (n_samples, n_features)
+            The original parameter space points to be transformed and evaluated.
+
+        Returns:
+        --------
+        kde_vals : array-like, shape (n_samples,)
+        The KDE values adjusted by the Jacobian of the transformations.
+        """
+        ### Transform points with transf, normalize, rescale
+        if self.input_transf is not None:
+            transf_points = transf.transform_data(points, self.input_transf)
+        else:
+            transf_points = points
+        #normalize
+        if self.stdize:
+            std_transf = ['stdize' for dim in self.ndim]
+            std_values_for_points = np.std(transf_points, axis=0)  # record the stds
+            std_points = transf.transform_data(transf_points, std_transf)
+        else:
+            std_values_for_points = None
+            std_points = transf_points
+
+        #rescale
+        if self.rescale is not None:
+            transf_data = transf.transform_data(std_points, self.rescale)
+        else:
+            transf_data = std_points
```

Comment:
This looks OK, but there should be some way of combining this set of transformations on the evaluate points with the prepare_data() method .. eg one could change prepare_data() into a method that takes some set of data points and outputs the transformed points and the stds (if standardization is done). 

Then this basic 'prepare some data' method could be called by other methods that either prepare the input training points or the evaluate points.

Reply:
Since this is a bit more complicated due to using self.stds in the standardization step, postpone this for later